### DISCUSSION TITLE
Are you supposed to close existing open database before calling importfromJson()?

### DISCUSSION THREAD

**Discussion by harikvpy** [capacitor-community/sqlite#567] [General]
I have a database service that keeps a connection to the database open in its `init()` method like this:

```
    this.db = await this.sqliteService.openDatabase(
      'mydb',
      false,
      'no-encryption',
      this.getDBVersion(),
      false
    );
    this.db = db;
```

`sqliteService` is an instance of the `SQLiteService` from the sample code.

What I find is that if you don't close this database and its connection, `importFromJson()` doesn't work. Especially when testing on a desktop with Jeep-sqlite. So what I do is 

```
    await this.db.close();
    await this.sqliteService.sqliteConnection.closeConnection('mydb', false);
```

and then call `importFromJson()` and then reopen the database.

Is this the correct approach? Is there anything that I'm missing?

**Comment by jepiqueau** [comment]
@harikvpy yes you need to close the connection do the importFromJson and create a new connection this is the process flow

**Reply by harikvpy** [reply]
Noted. Thank you.

**Comment by harikvpy** [comment]
Is this mentioned in the docs? May be it is, but I couldn't find it. If it isn't, it'll be nice to include it with a small code sample.

**Comment by jepiqueau** [comment]
@harikvpy you right, it is not clearly mentioned in the ImportExportJSON doc but if you look at the example in the link given [solide-vite-sqlite-app](https://github.com/jepiqueau/capacitor-solid-sqlite) you will see that the connection is close prior to use the importFromJson method. I will make it clear in the doc

**Comment by harikvpy** [comment]
I guess I was looking at the wrong samples. I downloaded [https://github.com/jepiqueau/blog-tutorials-apps.git](https://github.com/jepiqueau/blog-tutorials-apps.git) and searched for `importFromJson`, but couldn't find one. I'm using Angular BTW. Anyway, thanks for the quick response.
